odps-data-carrier/meta-processor/src/main/java/com/aliyun/odps/datacarrier/odps/datacarrier/HiveTypeTransformer.java

```diff
@@ -0,0 +1,157 @@
+package com.aliyun.odps.datacarrier.odps.datacarrier;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author: Jon (wangzhong.zw@alibaba-inc.com)
+ */
+public class HiveTypeTransformer {
+  //TODO: support odps1.0
+  // TODO: support hive.compatible
+  /**
+   * Numeric types
+   */
+  private static final String TINYINT = "TINYINT";
+  private static final String SMALLINT = "SMALLINT";
+  private static final String INT = "INT";
+  private static final String BIGINT = "BIGINT";
+  private static final String FLOAT = "FLOAT";
+  private static final String DOUBLE = "DOUBLE";
+  private static final String DECIMAL = "DECIMAL(\\([\\d]+,\\s*[\\d]+\\))?";
+
+  /**
+   * String types
+   */
+  private static final String STRING = "STRING";
+  private static final String VARCHAR = "VARCHAR(\\([\\d]+\\))";
+  private static final String CHAR = "CHAR(\\([\\d]+\\))";
+
+  /**
+   * Date and time types
+   */
+  private static final String TIMESTAMP = "TIMESTAMP";
+  private static final String DATE = "DATE";
+
+  /**
+   * Misc types
+   */
+  private static final String BOOLEAN = "BOOLEAN";
+  private static final String BINARY = "BINARY";
+
+  /**
+   * Complex types
+   */
+  private static final String ARRAY = "ARRAY<(.+)>";
+  private static final String MAP = "MAP<(.+)>";
+  private static final String STRUCT = "STRUCT<(.+)>";
+
+//  private static final String
+//  private static final String
+//  private static final String
+//  private static final String
+//  private static final String
+//  private static final String
+//  private static final String
+//  private static final String
+
+  public static String toOdpsType(String hiveType, String odpsVersion) {
+    hiveType = hiveType.toUpperCase();
+    if (hiveType.matches(TINYINT)) {
```

Comment:
it is ok for now, and this could be simplified by using a haspmap of String->lambda